infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java

```diff
@@ -145,38 +145,26 @@ private LogReplicationConfig generateConfig() {
                     .map(streamTag -> TableRegistry.getStreamIdForStreamTag(entry.getKey().getNamespace(), streamTag))
                     .collect(Collectors.toList()));
                 streamToTagsMap.put(streamId, tags);
+
+                // Remove this stream from the set of streams to drop
+                streamsToDrop.remove(CorfuRuntime.getStreamID(getFullyQualifiedTableName(entry.getKey())));
             }
 
             // TODO: Add other cases once the protobuf options for other subscribers are available
         });
 
-        // For each subscriber, add the Registry and Protobuf descriptor tables to the streams to replicate.
-        // Also construct the set of streams which must not be replicated.
-        Set<ReplicationSubscriber> subscribers = replicationSubscriberToStreamsMap.keySet();
-        Set<String> registryTableStreamNames = new HashSet<>();
-        registryTableEntries.forEach(entry -> registryTableStreamNames.add(getFullyQualifiedTableName(entry.getKey())));
-        Map<ReplicationSubscriber, Set<UUID>> subscriberToNonReplicatedStreamsMap = new HashMap<>();
+        // Add the Registry and Protobuf descriptor tables to the streams to replicate.
+        streamsToReplicate.add(getFullyQualifiedTableName(CORFU_SYSTEM_NAMESPACE,
+            TableRegistry.REGISTRY_TABLE_NAME));
+        streamsToReplicate.add(getFullyQualifiedTableName(CORFU_SYSTEM_NAMESPACE,
+            TableRegistry.PROTOBUF_DESCRIPTOR_TABLE_NAME));
 
-        for (ReplicationSubscriber subscriber : subscribers) {
-            Set<String> streamsToReplicate = replicationSubscriberToStreamsMap.get(subscriber);
+        ReplicationSubscriber subscriber = ReplicationSubscriber.getDefaultReplicationSubscriber();
 
-            String registryTableName = getFullyQualifiedTableName(CORFU_SYSTEM_NAMESPACE,
-                TableRegistry.REGISTRY_TABLE_NAME);
-            String protoTableName = getFullyQualifiedTableName(CORFU_SYSTEM_NAMESPACE,
-                TableRegistry.PROTOBUF_DESCRIPTOR_TABLE_NAME);
-
-            streamsToReplicate.add(registryTableName);
-            streamsToReplicate.add(protoTableName);
-
-            // Set of streams to drop
-            Set<UUID> set = registryTableStreamNames.stream().filter(stream -> !streamsToReplicate.contains(stream))
-                .map(CorfuRuntime::getStreamID).collect(Collectors.toSet());
-            subscriberToNonReplicatedStreamsMap.put(subscriber, set);
-        }
-
-        LogReplicationConfig config = new LogReplicationConfig(replicationSubscriberToStreamsMap,
-            subscriberToNonReplicatedStreamsMap, streamToTagsMap, serverContext);
-        return config;
+        Map<ReplicationSubscriber, LogReplicationConfig> configMap = new HashMap<>();
+        configMap.put(subscriber, new LogReplicationConfig(streamsToReplicate, streamsToDrop, streamToTagsMap,
+            serverContext));
+        return configMap;
```

Comment:
instead of returning the config, would be better to insert directly into subscriberToConfigMap.

This will also simplify the logic of getUpdatedConfig().